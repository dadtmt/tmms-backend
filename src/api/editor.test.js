import R from 'ramda'
import {
  getCrossroadsEdges,
  getCurrentCrossroadId,
  splitCrossroads,
  updateCrossroad,
  updateEditor
} from './editor'

describe('getCrossroadsEdges', () => {
  it('return empty edges if loading or no value', () => {
    const data = {}
    expect(getCrossroadsEdges(data)).toMatchSnapshot()
  })

  it(
    'return crossroads edges from the current (first of editors edges) editor',
    () => {
      const data = {
        viewer: {
          user: {
            editors: {
              edges: [{
                node: {
                  crossroads: {
                    edges: [
                      {
                        node: {
                          id: 'FIRST_CROSSROAD_ID'
                        }
                      },
                      {
                        node: {
                          id: 'SECOND_CROSSROAD_ID'
                        }
                      }
                    ]
                  },
                  id: 'EDITOR_ID'
                }
              }]
            }
          }
        }
      }
      expect(getCrossroadsEdges(data)).toMatchSnapshot()
    }
  )
})

describe('getCurrentCrossroadId', () => {
  it('return null if no crossroads', () => {
    const data = {}
    expect(getCurrentCrossroadId(data)).toMatchSnapshot()
  })

  it('return first crossroad id', () => {
    const data = {
      viewer: {
        user: {
          editors: {
            edges: [{
              node: {
                crossroads: {
                  edges: [
                    {
                      node: {
                        id: 'CURRENT_CROSSROAD_ID'
                      }
                    },
                    {
                      node: {
                        id: 'SECOND_CROSSROAD_ID'
                      }
                    },
                    {
                      node: {
                        id: 'THIRD_CROSSROAD_ID'
                      }
                    }
                  ]
                },
                id: 'EDITOR_ID'
              }
            }]
          }
        }
      }
    }
    expect(getCurrentCrossroadId(data)).toMatchSnapshot()
  })
})

describe('splitCrossroads', () => {
  it('return empty nodes', () => {
    const data = {}
    expect(splitCrossroads(data)).toMatchSnapshot()
  })
  it(
    'return currentCrossroad(first node) and lastCrossraods(the others)',
    () => {
      const data = {
        viewer: {
          user: {
            editors: {
              edges: [{
                node: {
                  crossroads: {
                    edges: [
                      {
                        node: {
                          id: 'CURRENT_CROSSROAD_ID'
                        }
                      },
                      {
                        node: {
                          id: 'SECOND_CROSSROAD_ID'
                        }
                      },
                      {
                        node: {
                          id: 'THIRD_CROSSROAD_ID'
                        }
                      }
                    ]
                  },
                  id: 'EDITOR_ID'
                }
              }]
            }
          }
        }
      }
      expect(splitCrossroads(data)).toMatchSnapshot()
    }
  )
})

describe('updateCrossroad', () => {
  it(
    'apply func update to the current (first of crossroads edges) crossroad',
    () => {
      const update = R.assoc(true, 'updated')
      const data = {
        viewer: {
          user: {
            editors: {
              edges: [{
                node: {
                  crossroads: {
                    edges: [{
                      node: {
                        id: 'CROSSROAD_ID'
                      }
                    }]
                  },
                  id: 'EDITOR_ID'
                }
              }]
            }
          }
        }
      }
      expect(updateCrossroad(update)(data)).toMatchSnapshot()
    }
  )
})

describe('updateEditor', () => {
  it('apply func update to the current (first of editors edges) editor', () => {
    const update = R.assoc(true, 'updated')
    const data = {
      viewer: {
        user: {
          editors: {
            edges: [{
              node: {
                id: 'EDITOR_ID'
              }
            }]
          }
        }
      }
    }
    expect(updateEditor(update)(data)).toMatchSnapshot()
  })
})
