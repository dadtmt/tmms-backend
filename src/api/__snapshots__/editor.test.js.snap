exports[`getCrossroadsEdges return crossroads edges from the current (first of editors edges) editor 1`] = `
Object {
  "edges": Array [
    Object {
      "node": Object {
        "id": "FIRST_CROSSROAD_ID",
      },
    },
    Object {
      "node": Object {
        "id": "SECOND_CROSSROAD_ID",
      },
    },
  ],
}
`;

exports[`getCrossroadsEdges return empty edges if loading or no value 1`] = `
Object {
  "edges": Array [],
}
`;

exports[`getCurrentCrossroadId return first crossroad id 1`] = `"CURRENT_CROSSROAD_ID"`;

exports[`getCurrentCrossroadId return null if no crossroads 1`] = `null`;

exports[`splitCrossroads return currentCrossroad(first node) and lastCrossraods(the others) 1`] = `
Object {
  "current": Object {
    "id": "CURRENT_CROSSROAD_ID",
  },
  "lasts": Array [
    Object {
      "node": Object {
        "id": "SECOND_CROSSROAD_ID",
      },
    },
    Object {
      "node": Object {
        "id": "THIRD_CROSSROAD_ID",
      },
    },
  ],
}
`;

exports[`splitCrossroads return empty nodes 1`] = `
Object {
  "current": null,
  "lasts": Array [],
}
`;

exports[`updateCrossroad apply func update to the current (first of crossroads edges) crossroad 1`] = `
Object {
  "viewer": Object {
    "user": Object {
      "editors": Object {
        "edges": Array [
          Object {
            "node": Object {
              "crossroads": Object {
                "edges": Array [
                  Object {
                    "node": Object {
                      "id": "CROSSROAD_ID",
                    },
                    "true": "updated",
                  },
                ],
              },
              "id": "EDITOR_ID",
            },
          },
        ],
      },
    },
  },
}
`;

exports[`updateEditor apply func update to the current (first of editors edges) editor 1`] = `
Object {
  "viewer": Object {
    "user": Object {
      "editors": Object {
        "edges": Array [
          Object {
            "node": Object {
              "id": "EDITOR_ID",
            },
            "true": "updated",
          },
        ],
      },
    },
  },
}
`;
